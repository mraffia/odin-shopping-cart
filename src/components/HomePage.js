import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/HomePage.css';

function HomePage() {
  return (
    <div className="home-container">
        <div className="home-subcontainer">
          <div className="home-texts-container">
            <h1 className="home-heading">Charms are gorgeous, mystical, one-of-a-kind accessories with a spark of power woven into their cores. Take a look at some of the Charms I have for sale!</h1>
            <p className="home-text">- Charm Lover Salubra</p>
            <Link to="/products" className="shop-link">
              <button className="shop-button" type="button" >Shop Now</button> 
            </Link>
          </div>
          <div className="home-image-container"></div>
        </div>
    </div>
  );
}

export default HomePage;
