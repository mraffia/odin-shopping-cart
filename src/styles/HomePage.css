.home-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100%;
    background-image: url('../images/salubra.png');
    background-size: cover;
    background-position: center;
}


.home-subcontainer {
    display: grid;
    grid-template-columns: 1fr 1fr;
    max-width: 1400px;
    animation: fadeInAnimation ease 2s;
    animation-iteration-count: 1;
    animation-fill-mode: forwards;
}

@keyframes fadeInAnimation {
    0% {
        opacity: 0;
    }
    100% {
        opacity: 1;
    }
}

.home-texts-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 40px;
    color: white;
}

.home-heading {
    margin-bottom: 0;
}

.home-text {
    margin-bottom: 36px;
}

.shop-button {
    padding: 16px;
    border: none;
    cursor: pointer;
    border-radius: 48px;
    width: 240px;
    font-weight: bold;
    font-size: 20px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    transition: transform .2s;
}

.shop-button:hover {
    transform: scale(1.06);
}

.home-image-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.home-image {
    max-width: 700px;
    border-radius: 36px;
}