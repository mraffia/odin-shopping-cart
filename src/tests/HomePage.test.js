import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter } from 'react-router-dom';
import HomePage from '../components/HomePage.js';

describe("HomePage component", () => {
    it("renders HomePage content", () => {
        const { container } = render(<HomePage />, {wrapper: BrowserRouter});
    
        expect(container).toMatchSnapshot();
    });
});