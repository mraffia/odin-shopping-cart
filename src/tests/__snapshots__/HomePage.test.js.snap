// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders HomePage content 1`] = `
<div>
  <div
    class="home-container"
  >
    <div
      class="home-subcontainer"
    >
      <div
        class="home-texts-container"
      >
        <h1
          class="home-heading"
        >
          Charms are gorgeous, mystical, one-of-a-kind accessories with a spark of power woven into their cores. Take a look at some of the Charms I have for sale!
        </h1>
        <p
          class="home-text"
        >
          - Charm Lover Salubra
        </p>
        <a
          href="/products"
        >
          <button
            class="shop-button"
            type="button"
          >
            Shop Now
          </button>
        </a>
      </div>
      <div
        class="home-image-container"
      />
    </div>
  </div>
</div>
`;
